bin/setup

```diff
@@ -7,6 +7,71 @@ fancy_echo() {
   printf "\\n$fmt\\n" "$@"
 }
 
+install_linux_deps() {
+  local LINUX_DEPS="automake autoconf autogen libtool gcc curl"
+  local flavor=`grep "^ID=" /etc/os-release | cut -d"=" -f 2`
+  echo $flavor
+  case $flavor in
+    debian|ubuntu)
+      sudo apt-get install -y $LINUX_DEPS libgmp-dev
```

Comment:
Will `apt-get` install the package even if it's already installed in the system? If so, it would be nice to check for the dependencies before we try to install them. That way the `bin/setup` script can be re-run multiple times without negative side effects. 

The same question applies to `yum` for fedora and centos.